### ISSUE TITLE
SFrameBuilder fails to report wrong length of input

### ISSUE THREAD

**Issue by DmitryZagursky** [apple/turicreate#1714]
When iterable of incorrect length is passed to the builder, it does not report error. But resulting sframe is unusable. In case of loading from disk it is only reported after you try to use the sframe.

```
sfb=turi.SFrameBuilder(column_types=[int,int])
sfb.append( (1,) )
sf=sfb.close()
>> RuntimeError: Assertion failed: (/builds/turi/turicreate/src/sframe/sarray_reader.hpp:277): sum==size()  [1 == 0]
```

**Comment by niknoproblems** [comment]
same error after group by

**Comment by Jarvi-Izana** [comment]
related to #1912. resolved in #2103. Be short, append with incorrect length iterable will end up with throwing an error to end user.
